op-chain-ops/deployer/pipeline/init.go

```diff
@@ -34,6 +35,37 @@ func Init(ctx context.Context, env *Env, artifactsFS foundry.StatDirFs, intent *
 		}
 	}
 
+	if intent.OPCMAddress != (common.Address{}) {
+		env.Logger.Info("using provided OPCM address, populating state", "address", intent.OPCMAddress.Hex())
+
+		if intent.ContractsRelease == "dev" {
+			env.Logger.Warn("using dev release with existing OPCM, this field will be ignored")
+		}
+
+		opcmContract := opcm.NewContract(intent.OPCMAddress, env.L1Client)
+		protocolVersions, err := opcmContract.ProtocolVersions(ctx)
+		if err != nil {
+			return fmt.Errorf("error getting protocol versions address: %w", err)
+		}
+		superchainConfig, err := opcmContract.SuperchainConfig(ctx)
+		if err != nil {
+			return fmt.Errorf("error getting superchain config address: %w", err)
+		}
+		env.Logger.Debug(
+			"populating protocol versions and superchain config addresses",
+			"protocolVersions", protocolVersions.Hex(),
+			"superchainConfig", superchainConfig.Hex(),
+		)
+
+		st.SuperchainDeployment = &state.SuperchainDeployment{
+			ProtocolVersionsProxyAddress: protocolVersions,
+			SuperchainConfigProxyAddress: superchainConfig,
+		}
```

Comment:
Comments here about the unused struct fields, and why they are left empty, would be nice here.